src/coordination/coordinator_state_manager.cpp

```diff
@@ -27,44 +27,61 @@ using nuraft::srv_state;
 using nuraft::state_mgr;
 
 namespace {
-constexpr std::string_view kServersKey = "servers";  // Key prefix for servers durability
+constexpr std::string_view kClusterConfigKey = "cluster_config";  // Key prefix for cluster_config durability
+
+constexpr std::string_view kServerStateKey = "server_state";  // Key prefix for server state durability
+constexpr std::string_view kVotedFor = "voted_for";
+constexpr std::string_view kTerm = "term";
+constexpr std::string_view kElectionTimer = "election_timer";
+
+constexpr std::string_view kStateManagerDurabilityVersionKey = "state_manager_durability_version";
+constexpr int kActiveStateManagerDurabilityVersion = 1;
+
 }  // namespace
 
-CoordinatorStateManager::CoordinatorStateManager(CoordinatorInstanceInitConfig const &config, LoggerWrapper logger)
-    : my_id_(static_cast<int>(config.coordinator_id)),
-      cur_log_store_(cs_new<CoordinatorLogStore>()),
+CoordinatorStateManager::CoordinatorStateManager(CoordinatorStateManagerConfig const &config, LoggerWrapper logger)
+    : my_id_(static_cast<int>(config.coordinator_id_)),
+      cur_log_store_(cs_new<CoordinatorLogStore>(config.durability_store_, logger)),
       logger_(logger),
-      kv_store_(config.durability_dir) {
+      state_manager_durability_(config.state_manager_durability_dir_) {
   auto const c2c =
-      CoordinatorToCoordinatorConfig{config.coordinator_id, io::network::Endpoint("0.0.0.0", config.bolt_port),
-                                     io::network::Endpoint{"0.0.0.0", static_cast<uint16_t>(config.coordinator_port)}};
-  my_srv_config_ = cs_new<srv_config>(config.coordinator_id, 0, c2c.coordinator_server.SocketAddress(),
-                                      nlohmann::json(c2c).dump(), false);  // non-resolved IP in SocketAddress
+      CoordinatorToCoordinatorConfig{config.coordinator_id_, io::network::Endpoint("0.0.0.0", config.bolt_port_),
```

Comment:
so we have separate RocksDB instances for CoordinatorLogStore and for StateManager?

Reply:
We talked yesterday about this, I think we can get on call for that again